### DISCUSSION TITLE
Unclear how to use `useOverlayTrigger` with `useMenu`

### DISCUSSION THREAD

**Discussion by bstro** [adobe/react-spectrum#3506] [Q&A] [answered]
I'm trying to build a menu with according to the docs, but I'm feeling a little confused.

The docs for [useOverlayTrigger](https://react-spectrum.adobe.com/react-aria/useOverlayTrigger.html#features) state:
> For menus, see [useMenu](https://react-spectrum.adobe.com/react-aria/useMenu.html), which builds on useOverlayTrigger and provides additional functionality specific to menus.

However, upon following the link to `useMenu`, there are no instructions for how one would construct a menu that uses positioning from `useOverlayTrigger`. It feels unclear how to compose all this together, or if they are meant to be composed.

Wondering if there is something I'm missing in the docs? Or an example somewhere?

Thanks!

**Comment by LFDanLu** [comment, marked as answer]
`useMenuTrigger` actually handles calling `useOverlayTrigger` for you so you can just follow the [example](https://react-spectrum.adobe.com/react-aria/useMenu.html#example), specifically the MenuButton portion. Something to note is that the [Popover](https://react-spectrum.adobe.com/react-aria/useMenu.html#popover) for `useMenu` is absolutely positioned with respect to the trigger button, so if you want to perform some additional positioning logic via `useOverlayPosition` you can take a look at its usage in the [useOverlayTrigger example](https://react-spectrum.adobe.com/react-aria/useOverlayTrigger.html#example).

**Reply by bstro** [reply]
Thanks!
